File: talks/2023/.NET Framework REST API/Server/Server/Domain/WeatherDaily.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Domain
{
    [Table("daily", Schema = "weather")]
    public class WeatherDaily
    {
        [Column(Order = 2), Key]
        public DateTime Date { get; set; }
        
        [Column(Order = 1), Key]
        public string Location { get; set; }

        [Column]
        public WeatherType Type { get; set; }

        [Column]
        public double Temperature { get; set; }

        [Column]
        public double Rain { get; set; }
    }
}

File: talks/2023/.NET Framework REST API/Server/Domain/WeatherDaily.fs
namespace Weather

open System
open System.ComponentModel.DataAnnotations
open System.ComponentModel.DataAnnotations.Schema
open Weather

[<Table("daily", Schema = "weather")>]
type WeatherDaily = 
    { 
        [<Column(Order = 2); Key>] Date: DateTime
        [<Column(Order = 1); Key>] Location: string
        [<Column>] Weather: WeatherType
        [<Column>] Temperature: double 
        [<Column>] Rain: double
    }


